tests/by-util/test_factor.rs

```diff
@@ -233,7 +233,7 @@ fn test_random_big() {
         // to generate an even split of this range, generate n-1 random elements
         // in the range, add the desired total value to the end, sort this list,
         // and then compute the sequential differences.
-        let mut f_bits = Vec::new();
+        let mut f_bits = Vec::with_capacity(n_factors);
```

Comment:
There's an additional `f_bits.push` directly after the `for`-loop (line 240):
```suggestion
        let mut f_bits = Vec::with_capacity(n_factors + 1);
```